modules/dnn/src/tensorrt/net_tensorrt.cpp

```diff
@@ -0,0 +1,873 @@
+    }
+}
+
+// remove file extention suffix
+static inline std::string removeFileSuffix(const std::string& filePath)
+{
+    size_t pos = filePath.find_last_of(".");
+
+    if (pos != std::string::npos)
+    {
+        return filePath.substr(0, pos);
+    }
+    else
+    {
+        return filePath;
+    }
+}
+
+static inline MatShape convertDim2Shape(const ::nvinfer1::Dims& dim)
+{
+    MatShape shape(dim.nbDims, 0);
+    memcpy(shape.data(), dim.d, dim.nbDims * sizeof(int));
+
+    return shape;
+}
+
+static inline ::nvinfer1::Dims convertShape2Dim(const MatShape& shape)
+{
+    ::nvinfer1::Dims dim;
+    dim.nbDims = shape.size();
+    memcpy(dim.d, shape.data(), shape.size() * sizeof(int));
+
+    return dim;
+}
+
+
+static std::vector<char> loadTimingCacheFile(const std::string inFileName)
+{
+    std::ifstream iFile(inFileName, std::ios::in | std::ios::binary);
+
+    if (!iFile)
+    {
+        CV_LOG_INFO(NULL, cv::String("[TensorRT EP] Could not read timing cache from: "+inFileName
+                            +". A new timing cache will be generated and written."));
+        return std::vector<char>();
+    }
+
+    iFile.seekg(0, std::ifstream::end);
+    size_t fsize = iFile.tellg();
+    iFile.seekg(0, std::ifstream::beg);
+    std::vector<char> content(fsize);
+    iFile.read(content.data(), fsize);
+    iFile.close();
+
+    return content;
+}
+
+static void saveTimingCacheFile(const std::string outFileName, const nvinfer1::IHostMemory* blob)
+{
+    std::ofstream oFile(outFileName, std::ios::out | std::ios::binary);
+
+    if (!oFile)
+    {
+        CV_LOG_INFO(NULL, cv::String("[TensorRT EP] Could not write timing cache to: "+outFileName));
+        return;
+    }
+
+    oFile.write((char*)blob->data(), blob->size());
+    oFile.close();
+}
+
+static std::string get_compute_capability(int device_id)
+{
+    cudaDeviceProp deviceProp;
+    cudaGetDeviceProperties(&deviceProp, device_id);
+
+    return "_" + std::to_string(deviceProp.major) + "." + std::to_string(deviceProp.minor);
+}
+
+void NetImplTrt::readNet(const String& model, const TrtConfig& configTRT)
+{
+    device_id_ = configTRT.deviceId;
+    std::string trt_model_filename = ""; // Contain the cache file name, when input is ONNX and need to save cache.
+    std::string path_to_write = "";
+
+    int GPU_count = 0;
+    cudaGetDeviceCount(&GPU_count);
+    CV_Assert(device_id_ >= 0 && device_id_ < GPU_count && "The device id does not exist!");
+
+    // TODO support multi-GPU
+    cudaSetDevice(device_id_); // set specific GPU device for TensorRT backend.
+
+    this->compute_capability_ = get_compute_capability(device_id_);
+
+    const std::string modelExt = model.substr(model.rfind('.') + 1);
+    bool is_trt_model = false;
+
+    if (configTRT.useCache)
+    {
+        if (!configTRT.cachePath.empty())
+        {
+            path_to_write = configTRT.cachePath + "/";
+        }
+        else
+        {
+            path_to_write = removeFileName(model) + "/";
```

Comment:
Doe it mean that cache relies on ONNX file base name? Maybe something like `OPENCV_DNN_TRT_SERIALIZE` similar to [OPENCV_DNN_IE_SERIALIZE](https://github.com/opencv/opencv/blob/5b3f33dd11ae42316378de6b3bcf57118ebc19ab/modules/dnn/src/ie_ngraph.cpp#L30) can be more stable?

Reply:
Thanks, I will take a look at it.